mypy/checker.py

```diff
@@ -1290,8 +1290,8 @@ def check_override(self, override: FunctionLike, original: FunctionLike,
             fail = True
 
         if isinstance(original, CallableType) and isinstance(override, CallableType):
-            if (isinstance(original.definition, FuncItem) and
-                    isinstance(override.definition, FuncItem)):
+            if (isinstance(original.definition, FuncBase) and
```

Comment:
Note that this PR (in particular because of this change) may have conflicts with my type aliases refactoring. In turned out that the `type_override` (that I killed) was also used for some overload hacks.

Reply:
I'm not sure I'm completely following -- is there now going to be some major difference between FuncItem and FuncBase?

(I also ended up streamlining this code to handle the "overriding a classmethod with an instance method" case more cleanly -- not sure if that change makes the potential conflict better or worse.)